### DISCUSSION TITLE
SRI - Subresource Integrity

### DISCUSSION THREAD

**Discussion by andygmorris** [vercel/next.js#15964] [Help]
For one of our developer friends...
How can they implement SRI upon npm run build in a nextjs app so that integrity is added to the script line?
Something like <script src="{scriptname}" integrity="{hash}" crossorigin="anonymous"> when building the app.
Any pointers apprciated, and I'll pass them on.

**Comment by morgansson** [comment]
I'm currently facing this issue, I need SRI and my app is built using Nextjs.

I believe SRI is the future and the decentralized internet that's ahead of us requires that top frameworks like Nextjs adapt to support this technology.

I'm thinking of doing some kind of parser to apply the SRI tags after the build is complete, as a separate task. I don't know how to actually fix this in any other way with Nextjs 12, as I'm using the swc compiler. Any ideas would be great.

Thanks a lot to everyone.

**Comment by JBGoldberg** [comment]
Is there any evolution in this issue on Nextjs 14?

**Comment by jpcors** [comment]
It looks like there is an experimental flag for SRI but I can't seem to get it to work..

https://github.com/vercel/next.js/pull/39729
